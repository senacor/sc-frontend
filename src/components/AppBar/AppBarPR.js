import React, { useState, Fragment } from 'react';
import Typography from '@material-ui/core/Typography';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import { withStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Hidden from '@material-ui/core/Hidden';
import Drawer from '@material-ui/core/Drawer';
import { injectIntl } from 'react-intl';

import Sidebar from '../sidebar/Sidebar';
import LanguageButton from './LanguageButton';

const drawerWidth = 270;
const appBarHeight = 64;
const styles = theme => ({
  appBar: {
    height: appBarHeight,
    marginLeft: drawerWidth,
    [theme.breakpoints.up('lg')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  navIconHide: {
    [theme.breakpoints.up('lg')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth,
    [theme.breakpoints.up('lg')]: {
      position: 'fixed'
    }
  },
  languageButton: {
    color: theme.palette.contrastText,
    marginLeft: 'auto'
  }
});

const AppBarPR = ({ classes, intl, theme }) => {
  const [drawerOpen, setDrawerOpen] = useState(false);

  const handleDrawerToggle = () => {
    setDrawerOpen(!drawerOpen);
  };

  return (
    <Fragment>
      <AppBar className={classes.appBar}>
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label={intl.formatMessage({
              id: 'appbar.drawer'
            })}
            onClick={handleDrawerToggle}
            className={classes.navIconHide}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" color="inherit" noWrap>
            {intl.formatMessage({
              id: 'appbar.portal'
            })}
          </Typography>
          <LanguageButton languageButtonClassName={classes.languageButton} />
        </Toolbar>
      </AppBar>
      <Hidden lgUp>
        <Drawer
          variant="temporary"
          anchor={theme.direction === 'rtl' ? 'right' : 'left'}
          open={drawerOpen}
          onClose={handleDrawerToggle}
          classes={{
            paper: classes.drawerPaper
          }}
          ModalProps={{
            keepMounted: true // Better open performance on mobile.
          }}
        >
          <Sidebar />
        </Drawer>
      </Hidden>
      <Hidden mdDown implementation="css">
        <Drawer
          variant="permanent"
          open
          classes={{
            paper: classes.drawerPaper
          }}
        >
          <Sidebar />
        </Drawer>
      </Hidden>
    </Fragment>
  );
};

export default injectIntl(withStyles(styles, { withTheme: true })(AppBarPR));
