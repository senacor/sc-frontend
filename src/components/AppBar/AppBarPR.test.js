import React from 'react';
import { shallowWithIntl } from 'enzyme-react-intl';

import AppBarPR from './AppBarPR';

describe('AppBarPR', () => {
  it('should render correctly', () => {
    const wrapper = shallowWithIntl(<AppBarPR />);
    expect(wrapper).toMatchSnapshot();
  });
});
