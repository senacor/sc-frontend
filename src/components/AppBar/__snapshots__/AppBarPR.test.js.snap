// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBarPR should render correctly 1`] = `
<WithStyles(AppBarPR)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
/>
`;
